### PR TITLE
chore(DIA-942): test auth2 login welcome step

### PR DESCRIPTION
**Pull request by iskounen**
This PR partially resolves [DIA-942] 

### Description

This PR adds tests for the behavior in the LoginWelcomeStep of the new auth flow. It was very difficult to mock the navigation, but now that I've figured it out for this step I hope the following steps are easier.

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- Added tests for LoginWelcomeStep

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[DIA-942]: https://artsyproduct.atlassian.net/browse/DIA-942?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Dev changes (Added tests for LoginWelcomeStep - iskounen)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 17dad365cbbef909a856c12b6a55d44c33f18eaf
</p>

File: src/app/Scenes/Onboarding/Auth2/scenes/LoginWelcomeStep.tsx [hunk 1]
```diff
@@ -130,6 +130,7 @@ const LoginWelcomeStepForm: React.FC = () => {
       <Text variant="sm-display">Sign up or log in</Text>
 
       <Input
+        accessibilityHint="Enter your email address"
```
Comment (damassi) [inline src/app/Scenes/Onboarding/Auth2/scenes/LoginWelcomeStep.tsx:133]: 🙏

File: src/app/Scenes/Onboarding/Auth2/scenes/__tests__/LoginWelcomeStep.tests.tsx [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+  const mockUseAuthNavigation = useAuthNavigation as jest.Mock
+  const mockUseRecaptcha = useRecaptcha as jest.Mock
+  const mockOSMajorVersion = osMajorVersion as jest.Mock
+
+  beforeEach(() => {
+    /**
+     * This component involves animations, so we need to mock timers to ensure that the animations
+     * are completed before making assertions.
+     */
+    jest.useFakeTimers()
```
Comment (damassi) [inline src/app/Scenes/Onboarding/Auth2/scenes/__tests__/LoginWelcomeStep.tests.tsx:33]: TIL

**Review by damassi — APPROVED** [review]
These tests are very nicely done 👍 thanks for the code comment regarding animation, too; will def save us some time in the future.

**Pull request merged**